BlueprintUI/Sources/AttributedText/AttributedTextKey.swift

```diff
@@ -0,0 +1,97 @@
+import UIKit
+
+/// Define `AttributeText` keys using this protocol. Keys must have an attribute name
+/// and an associated type for the attribute.
+///
+/// After defining a key, enable dynamic member access to it by extending `TextAttributeContainer`
+/// with a property for getting and setting setting the value. This property should generally be optional,
+/// since the text may not have that property defined.
+///
+public protocol AttributedTextKey {
+    associatedtype Value: Equatable
+    static var name: NSAttributedString.Key { get }
+}
+
+// MARK: - Built-in attribute keys
+
+// MARK: Font
+
+public enum FontKey: AttributedTextKey {
+    public typealias Value = UIFont
+    public static var name: NSAttributedString.Key { .font }
+}
+
+extension TextAttributeContainer {
+    public var font: UIFont? {
+        get { self[FontKey.self] }
+        set { self[FontKey.self] = newValue }
+    }
+}
+
+// MARK: Color
+
+public enum ColorKey: AttributedTextKey {
+    public typealias Value = UIColor
+    public static var name: NSAttributedString.Key { .foregroundColor }
+}
+
+extension TextAttributeContainer {
+    public var color: UIColor? {
+        get { self[ColorKey.self] }
+        set { self[ColorKey.self] = newValue }
+    }
+}
+
+// MARK: Kern
+
+public enum KernKey: AttributedTextKey {
+    public typealias Value = CGFloat
+    public static var name: NSAttributedString.Key { .kern }
```

Comment:
For the built-in keys, I think we might want to expose tracking instead of kerning, via `kCTTrackingAttributeName as NSAttributedString.Key`.

Check out [this comment thread where we added it to POSUI for context](https://github.com/squareup/ios-register/pull/49844#discussion_r554402886).

Reply:
Done